src/Reader/StandaloneExtensionManager.php

```diff
@@ -49,4 +49,30 @@ public function get($extension)
         $class = $this->extensions[$extension];
         return new $class();
     }
+
+    /**
+     * Add an extension.
+     *
+     * @param string $name
+     * @param string $class
+     */
+    public function add($name, $class)
+    {
+        $this->extensions[$name] = $class;
```

Comment:
No validation what is added to the manager?

Reply:
good point. `$class` should be checked to be instance of `Zend\Feed\Reader\Extension\AbstractEntry` or `Zend\Feed\Reader\Extension\AbstractFeed`

Reply:
I wondered about that - $class can be a string (it was in the components until I changed it), but it might be the name of something that doesn't exist yet, so I don't know that it can be validated at this point.

Reply:
with check it will be autoloaded when it is registered. Plugin manager validates on creation.